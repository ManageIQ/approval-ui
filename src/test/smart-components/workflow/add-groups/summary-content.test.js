import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SummaryContent from '../../../../smart-components/workflow/add-groups/summary-content';
import rendererContext from '@data-driven-forms/react-form-renderer/dist/cjs/renderer-context';

describe('<SummaryContent />', () => {
  let formData;

  beforeEach(() => {
    formData = {
      name: 'Group summary name',
      description: 'Group summary description',
      wfGroups: []
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<rendererContext.Provider value={ { formOptions: {
      getState: () => ({
        values: formData
      })
    }} }>
      <SummaryContent />
    </rendererContext.Provider>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
