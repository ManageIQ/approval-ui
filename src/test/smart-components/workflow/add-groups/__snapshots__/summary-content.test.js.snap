// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryContent /> should render correctly with groups 1`] = `
<SummaryContent>
  <Stack
    gutter="md"
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Title
            size="xl"
          >
            <h1
              className="pf-c-title pf-m-xl"
            >
               Review 
            </h1>
          </Title>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Stack
            gutter="md"
          >
            <div
              className="pf-l-stack pf-m-gutter"
            >
              <StackItem>
                <div
                  className="pf-l-stack__item"
                >
                  <TextContent>
                    <div
                      className="pf-c-content"
                    >
                      <Text
                        className="pf-u-mb-0"
                        component="small"
                      >
                        <small
                          className="pf-u-mb-0"
                          data-pf-content={true}
                        >
                          Review and confirm your inputs. If there is anything incorrect, click Back and revise.
                        </small>
                      </Text>
                    </div>
                  </TextContent>
                </div>
              </StackItem>
              <StackItem>
                <div
                  className="pf-l-stack__item"
                >
                  <Grid
                    gutter="md"
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={2}
                      >
                        <div
                          className="pf-l-grid__item pf-m-2-col"
                        >
                          <Text
                            className="pf-u-mb-0"
                            component="small"
                          >
                            <small
                              className="pf-u-mb-0"
                              data-pf-content={true}
                            >
                              Name
                            </small>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={10}
                      >
                        <div
                          className="pf-l-grid__item pf-m-10-col"
                        >
                          <Text
                            className="pf-u-mb-md"
                            component="p"
                          >
                            <p
                              className="pf-u-mb-md"
                              data-pf-content={true}
                            >
                              Group summary name
                            </p>
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                  <Grid
                    gutter="md"
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={2}
                      >
                        <div
                          className="pf-l-grid__item pf-m-2-col"
                        >
                          <Text
                            className="pf-u-mb-0"
                            component="small"
                          >
                            <small
                              className="pf-u-mb-0"
                              data-pf-content={true}
                            >
                              Description
                            </small>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={10}
                      >
                        <div
                          className="pf-l-grid__item pf-m-10-col"
                        >
                          <Text
                            className="pf-u-mb-md"
                            component="p"
                          >
                            <p
                              className="pf-u-mb-md"
                              data-pf-content={true}
                            >
                              Group summary description
                            </p>
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                  <Grid
                    gutter="md"
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={2}
                      >
                        <div
                          className="pf-l-grid__item pf-m-2-col"
                        >
                          <Text
                            className="pf-u-mb-0"
                            component="small"
                          >
                            <small
                              className="pf-u-mb-0"
                              data-pf-content={true}
                            >
                              Groups
                            </small>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={10}
                      >
                        <div
                          className="pf-l-grid__item pf-m-10-col"
                        >
                          <Text
                            className="pf-u-mb-md"
                            component="p"
                          >
                            <p
                              className="pf-u-mb-md"
                              data-pf-content={true}
                            >
                              Group 1
                            </p>
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                  <Grid
                    gutter="md"
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={2}
                      >
                        <div
                          className="pf-l-grid__item pf-m-2-col"
                        >
                          <Text
                            className="pf-u-mb-0"
                            component="small"
                          >
                            <small
                              className="pf-u-mb-0"
                              data-pf-content={true}
                            />
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={10}
                      >
                        <div
                          className="pf-l-grid__item pf-m-10-col"
                        >
                          <Text
                            className="pf-u-mb-md"
                            component="p"
                          >
                            <p
                              className="pf-u-mb-md"
                              data-pf-content={true}
                            >
                              Group 2
                            </p>
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                </div>
              </StackItem>
            </div>
          </Stack>
        </div>
      </StackItem>
    </div>
  </Stack>
</SummaryContent>
`;

exports[`<SummaryContent /> should render correctly without groups 1`] = `
<SummaryContent>
  <Stack
    gutter="md"
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Title
            size="xl"
          >
            <h1
              className="pf-c-title pf-m-xl"
            >
               Review 
            </h1>
          </Title>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Stack
            gutter="md"
          >
            <div
              className="pf-l-stack pf-m-gutter"
            >
              <StackItem>
                <div
                  className="pf-l-stack__item"
                >
                  <TextContent>
                    <div
                      className="pf-c-content"
                    >
                      <Text
                        className="pf-u-mb-0"
                        component="small"
                      >
                        <small
                          className="pf-u-mb-0"
                          data-pf-content={true}
                        >
                          Review and confirm your inputs. If there is anything incorrect, click Back and revise.
                        </small>
                      </Text>
                    </div>
                  </TextContent>
                </div>
              </StackItem>
              <StackItem>
                <div
                  className="pf-l-stack__item"
                >
                  <Grid
                    gutter="md"
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={2}
                      >
                        <div
                          className="pf-l-grid__item pf-m-2-col"
                        >
                          <Text
                            className="pf-u-mb-0"
                            component="small"
                          >
                            <small
                              className="pf-u-mb-0"
                              data-pf-content={true}
                            >
                              Name
                            </small>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={10}
                      >
                        <div
                          className="pf-l-grid__item pf-m-10-col"
                        >
                          <Text
                            className="pf-u-mb-md"
                            component="p"
                          >
                            <p
                              className="pf-u-mb-md"
                              data-pf-content={true}
                            >
                              Group summary name
                            </p>
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                  <Grid
                    gutter="md"
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={2}
                      >
                        <div
                          className="pf-l-grid__item pf-m-2-col"
                        >
                          <Text
                            className="pf-u-mb-0"
                            component="small"
                          >
                            <small
                              className="pf-u-mb-0"
                              data-pf-content={true}
                            >
                              Description
                            </small>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={10}
                      >
                        <div
                          className="pf-l-grid__item pf-m-10-col"
                        >
                          <Text
                            className="pf-u-mb-md"
                            component="p"
                          >
                            <p
                              className="pf-u-mb-md"
                              data-pf-content={true}
                            >
                              Group summary description
                            </p>
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                </div>
              </StackItem>
            </div>
          </Stack>
        </div>
      </StackItem>
    </div>
  </Stack>
</SummaryContent>
`;
